lib/faraday/adapter_registry.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'monitor'
+
+module Faraday
+  # AdapterRegistry registers adapter class names so they can be looked up by a
+  # String or Symbol name.
+  class AdapterRegistry
+    def initialize
+      @lock = Monitor.new
+      @constants = nil
+    end
+
+    def get(name)
+      klass = @constants && @constants[name]
+      return klass if klass
+
+      klass =
+        if name.respond_to?(:constantize)
+          name.constantize
```

Comment:
This line is untested by the existing unit tests.

Is this a useful feature?
I would think that `Object.const_get(name)` should be enough, but this may just be me not appreciating the benefits of `constantize`

Reply:
Good point here, they should do the same thing but according to `benchmark`, Rails `#constantize` is around 4 times slower than `const_get`. Plus the former is only available under `active_support`, while the latter is always available. 

Is there a particular reason for not simply using `const_get`?

Reply:
It was used because `Object.const_get` was not able to traverse module namespaces (ex: `Object.const_get('Faraday::Connection')`. Apparently that was changed some time in the last decade, so let's 🔥 constantize down!